### PR TITLE
ci: verify that test course JSON files are not out of date. 

### PR DESCRIPTION
**Pull request by gdomingu**
## Description

Adds a script and updates semaphore to check if the Course Test Json Files are up to date.

**Is this related to any open issue(s)?**

fixes https://github.com/kantord/LibreLingo/issues/1447

**Is there anything you need help with to get this PR merged?**

I need help resolving CI errors:
```
ImportError: cannot import name 'AudioSettings' from 'librelingo_types' (/home/semaphore/.pyenv/3.8/lib/python3.8/site-packages/librelingo_types/__init__.py)
```

## Checklist

Don't worry if you haven't done everything on this checklist. Simply create your PR, and make sure everything is done later,
or ask for help if you don't know how to do it.

- [ ] The CI is passing (This will need to go in first for CI to pass https://github.com/kantord/LibreLingo/pull/1613)
- [x] I tested my implementation manually
- [ ] I wrote new tests to test my code, or updated existing tests to test my changes

### CODE REVIEW

**Comment by gdomingu** [comment]
@kantord any advice as to why I might be getting 
```
ImportError: cannot import name 'AudioSettings' from 'librelingo_types' (/home/semaphore/.pyenv/3.8/lib/python3.8/site-packages/librelingo_types/__init__.py)
```
In CI? I didn't see it locally.

File: scripts/exportAllYamlCourses.sh [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
+ANY_FAILED="0"
 for d in ./courses/*/ ; do
 		d=$(echo $d | sed 's/[^/]*\/[^/]*\///' | sed 's/\///')
 		./scripts/exportYamlCourse.sh $d
+		if [ "$?" -eq "1" ]; then
+			ANY_FAILED="1"
+		fi
 done
 
+if [ "$ANY_FAILED" -eq "1" ]; then
```
Comment (kantord) [inline scripts/exportAllYamlCourses.sh:12]: This is a really cool new feature but right now it will always fail, right? So it might be a good idea to somehow separate this into a separate MR, and merge it once we've dealt with that issue (for example somehow temporarily disabled the courses that are breaking)

File: scripts/verifyTestCourseJsonFiles.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+set -e
+
+yarn exportAllCourses -B
```
Comment (kantord) [inline scripts/verifyTestCourseJsonFiles.sh:4]:
```suggestion
yarn exportAllCourses
```
 Or what was that about?

File: scripts/verifyTestCourseJsonFiles.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+if [ "$?" -eq "0" ]; then
+	echo "🎉 Amazing, test course JSON files are up to date"
+else
+	echo "😿 Outdated test course JSON files detected."
+	echo "The following files were outdated:"
+	git diff --name-only
+	echo
+	echo "See the full list of changes:"
+	git diff
+	echo "Run the following script to automatically update the readme files:"
```
Comment (kantord) [inline scripts/verifyTestCourseJsonFiles.sh:16]:
```suggestion
	echo "Run the following script to automatically update the JSON files:"
```

**Review by kantord — CHANGES_REQUESTED** [review]

**Comment by kantord** [comment]
> @kantord any advice as to why I might be getting 
> ```
> ImportError: cannot import name 'AudioSettings' from 'librelingo_types' (/home/semaphore/.pyenv/3.8/lib/python3.8/site-packages/librelingo_types/__init__.py)
> ```
> In [CI](https://kantord.semaphoreci.com/jobs/3e2d39c0-50c1-4001-8465-4e63f38974ce)? I didn't see it locally. 

I think the root cause is that there's s mistake in how exportAllCourses is implemented. Right now it goes into the folder of the Python package and tries to run it from there, but that doesn't work out because of the way poetry manages dependencies. Basically it doesn't have access to the correct virtualenv, as poetry always used the virtualenv that belongs to the closes poetry lockfile. But that virtualenv is not being installed. Quick and dirty fix is to run `poetry install` before exporting courses. Real fix is to not `cd` away from the project root folder

File: scripts/exportAllYamlCourses.sh (hunk 1 above)
Comment (gdomingu) [reply scripts/exportAllYamlCourses.sh:12]: Sounds good.

**Comment by gdomingu** [comment]
@kantord Thank you for the help! I have updated and it seems to be working now (CI Failing because the files are out of date).
Screen Shot 2021-09-19 at 9 05 58 AM

**Review by kantord — APPROVED** [review]
looks good to me, let's merge this: https://github.com/kantord/LibreLingo/pull/1613

and then we can merge this

**Comment by kantord** [comment]
Thanks @gdomingu! 💯 🎉 Let's see if it really works on the `main` branch!
